### ISSUE TITLE
Sidekiq.strict_args! rejects Hash with symbolic keys

### ISSUE THREAD

**Issue by MacksMind** [mperham/sidekiq#5221]
This line flag Hash args with symbolic keys as unsafe because the parsed JSON results in string keys. Would you consider a PR to preprocess such hashes before the test?
https://github.com/mperham/sidekiq/blob/4ce86cda9e968de0d3aa198839c2dd0ac4c4f17b/lib/sidekiq/job_util.rb#L61

**Comment by mperham** [comment]
The Best Practices specifically say not to use Symbols. You can call `stringify_keys` on the Hash first though.

**Comment by MacksMind** [comment]
That's on me for just skimming then. 🤦‍♂️ 

Thanks!

**Comment by mperham** [comment]
I’ve updated the Wiki page to document that helper method. Thanks for the issue, always room for improvement!

**Comment by bfad** [comment]
Can I ask what the logic is for having to call `stringify_keys` when passing a hash to `perform_async`? 
1. The call to `JSON.dump` will do that for you
2. I don't see it as preventing coding mistakes — if you use the hash with symbols in the code that passes it to the worker, you're still likely to try and use it as symbols inside the `perform` function. (After all, you're only stringifying it for `perform_async` and all my unit tests of `perform` pass when I call it with a symbol.)

I guess I don't think the inconvenience is worth the cost. I love that this check exists for complex data types, but I'm more likely to turn this off then go through my large codebase and annotate every instance I pass a hash with `stringify_keys`.

**Comment by mperham** [comment]
@bfad Do you understand the purpose of `strict_args!`? Your point (2) is wrong: a Symbol will go in and a String will come out. That's how JSON works.

**Comment by bfad** [comment]
@mperham Yes, what I'm saying is that JSON.dump will stringify the hash keys for me, so why should I have to do it when I call `perform_async`?

I understand that my second point was unclear, but the point is that I don't think this will help prevent coding mistakes. I understand that if I'm using the hash with symbols outside the call to `perform_async`, I'm more likely to use symbols for the hash passed to the worker too, even though the JSON serialization and back will stringify those keys. Let's say I've made that mistake and even written unit tests for the `perform` method using symbols too. These will all pass and not show this warning or raise this error since they're of the form `Worker.new.perform(...)`. Now let's say I see the warning message in my other tests and find out that hashes are supported as long as I call Rails's `Hash#stringify_keys` when I call `Worker.perform_async`. I add it when calling `Worker.perform_async` and the message is gone. I may still think everything works fine now even though I'm still using symbols inside `perform` as my tests are all still passing. People will still have to notice that things aren't working, and they'll still need to figure out why.

I hope I articulated that second point more clearly.

Essentially, I think that automatically converting a hash's symbols to strings without having to call `stringify_keys` everywhere is a developer convenience feature.
